tests/run-macros/quote-matcher-inference/Macro_1.scala

```diff
@@ -0,0 +1,15 @@
+import scala.quoted._
+
+
+object Macros {
+
+  inline def g(inline x: Unit): Unit = ${impl('x)}
+
+  private def impl(x: Expr[Any])(using QuoteContext): Expr[Any] = {
+    x match
+      case '{ println(f($y)) } => y
+  }
+
+}
+
+def f[T](x: T): T = x
```

Comment:
`flipBottom` only special-cases Nothing, but the same issue could reoccur with a different bound like `>: Null`. I'm not sure how the example above work anyway, when you write:
```scala
case '{ println(f($y)) } => y
```
shouldn't this be expanded to:
```scala
case '{ println(f[$t]($y)) } => y
```
And that way, any `println(f[...](...))` call will match, no matter the type argument ?

Reply:
That is the current workaround, but it seems hard to expand it while typing it. At some point, this should be tried.

Reply:
This is just meant as a patch for a relatively common code pattern. I'm working on the side on implementing the whole pattern desugaring/typing logic which might help us with those tricky expansions.

Reply:
That would be great!